src/client/voice/VoiceConnection.js

```diff
@@ -447,12 +447,12 @@ class VoiceConnection extends EventEmitter {
      */
     if (this.status === VoiceStatus.CONNECTED) {
       this.emit('speaking', user, speaking);
-      if (!speaking) {
+      if (!speaking.has(Speaking.FLAGS.SPEAKING)) {
         this.receiver.packets._stoppedSpeaking(user_id);
       }
     }
 
-    if (guild && user && old !== speaking) {
+    if (guild && user && old.equals(speaking)) {
```

Comment:
need to check that it doesn't equal, so missed the `!` from my comment.

Reply:
As you have probably seen, sorted now. (Sorry about that, apparently my testing does not care about that code path). Cheers!